pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ClusterTest.java

```diff
@@ -125,6 +128,15 @@ protected void startBrokers(int numBrokers, int basePort, String zkStr)
       brokerStarter.start();
       _brokerStarters.add(brokerStarter);
     }
+    _brokerBaseApiUrl = "http://localhost:" + _brokerPorts.get(0);
+  }
+
+  protected int getBrokerPort(int index) {
```

Comment:
Do not pass in index but return a random broker port?

Reply:
this actually keeps the old behavior, updated the logic to return a random port.